services/github/github-issues-search.service.js

```diff
@@ -0,0 +1,112 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('joi')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+
+const issueCountSchema = Joi.object({
+  data: Joi.object({
+    search: Joi.object({
+      issueCount: nonNegativeInteger,
+    }).required(),
+  }).required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  query: Joi.string().required(),
+}).required()
+
+class BaseGithubIssuesSearch extends GithubAuthV4Service {
+  static category = 'issue-tracking'
+  static defaultBadgeData = { label: 'query', color: 'informational' }
+
+  static render({ issueCount }) {
+    return { message: metric(issueCount) }
+  }
+
+  async fetch({ query }) {
+    const data = await this._requestGraphql({
+      query: gql`
+        query($query: String!) {
+          search(query: $query, type: ISSUE) {
```

Comment:
will the `type: ISSUE` filter out PRs? If so will we need a way to make this conditional to enable our existing badge service classes to be able to leverage?

Reply:
`type: ISSUE` includes issues and PRs. If your query includes `is:issue` you just get issues. If your query includes `is:pr` you get just PRs. If it includes neither, you get both.